tests/vu/lower/fdivdelay.cpp

```diff
@@ -0,0 +1,243 @@
+#include <common-ee.h>
+#include "../test_runner.h"
+
+static VU::LowerOp DIV_adapter(VU::Field field, VU::Reg reg) {
+	using namespace VU;
+	return DIV(FIELD_W, VF00, field, reg);
+}
+
+static VU::LowerOp RSQRT_adapter(VU::Field field, VU::Reg reg) {
+	using namespace VU;
+	return RSQRT(FIELD_W, VF00, field, reg);
+}
+
+class FdivDelayTestRunner : public TestRunner {
+public:
+	FdivDelayTestRunner(int vu) : TestRunner(vu) {
+	}
+
+	typedef VU::LowerOp (*FdivFunction)(VU::Field field, VU::Reg reg);
+
+	void PerformSingleWaitQ(const char *name, FdivFunction func) {
+		using namespace VU;
+
+		printf("%s single waitq:\n", name);
+
+		Reset();
+
+		//Prepare registers
+		WrImm(0x40800000);
+		Wr(MULi(DEST_W, VF01, VF00));
+		
+		//Op
+		Wr(func(FIELD_W, VF01));
+		Wr(WAITQ());
+		Wr(MULq(DEST_XYZW, VF01, VF00));
+		
+		Execute();
+
+		printf("  VF01: "); PrintRegister(VF01, true);
+	}
+
+	void PerformMultipleWaitQ(const char *name, FdivFunction func) {
+		using namespace VU;
+
+		printf("%s multiple waitq:\n", name);
+
+		Reset();
+
+		//Prepare registers
+		WrImm(0x40800000);
+		WrImm(MULi(DEST_W, VF01, VF00), 0x41800000);
+		Wr(MULi(DEST_W, VF02, VF00));
+		
+		//Op
+		Wr(func(FIELD_W, VF01));
+		Wr(WAITQ());
+		Wr(MULq(DEST_XYZW, VF01, VF00));
+		Wr(func(FIELD_W, VF02));
+		Wr(WAITQ());
+		Wr(MULq(DEST_XYZW, VF02, VF00));
+
+		Execute();
+
+		printf("  VF01: "); PrintRegister(VF01, true);
+		printf("  VF02: "); PrintRegister(VF02, true);
+	}
+
+	void PerformSingleDelayed(const char *name, FdivFunction func) {
+		using namespace VU;
+
+		printf("%s single delayed:\n", name);
+
+		Reset();
+
+		//Set Q to 1.0f
+		Wr(DIV(FIELD_W, VF00, FIELD_W, VF00));
+		Wr(WAITQ());
+
+		//Prepare registers
+		WrImm(0x40800000);
```

Comment:
In this case I would prefer 4.0f.  It's overloaded for floats and in this case would make things clearer.

-[Unknown]